gms/gossiper.cc

```diff
@@ -1220,8 +1215,16 @@ int64_t gossiper::get_endpoint_downtime(inet_address ep) const noexcept {
 // It is called from failure_detector
 future<> gossiper::convict(endpoint_id node) {
     auto permit = co_await lock_endpoint(node, null_permit_id);
-    auto state = get_endpoint_state_ptr(node);
-    if (!state || !is_alive(node)) {
+    auto state = get_endpoint_state_ptr(node.host_id);
+    if (!state) {
+        state = get_endpoint_state_ptr(node.addr);
```

Comment:
Why not keep this logic inside `get_endpoint_state_ptr(node)`?

Reply:
hmm, it's probably a bad idea to always fall back to IP. Why do we fall back to IP here?